Make ModuleArgsParser more understandable

willthames opened the pull request:
Both comments and method names for handling new/old
style parameters were switched around

Made comments and method names reflect actual code paths
taken.

willthames committed:
Make ModuleArgsParser more understandable

Both comments and method names for handling new/old
style parameters are switched around

Made comments and method names reflect actual code paths
taken.

bcoca commented on lib/ansible/modules/extras:
```diff
@@ -1 +1 @@
-Subproject commit f798240f436a16a828f48759bbd176b6bccdfe75
+Subproject commit b20cb4dd71425a48e68e0af06039afd2a410cb13
```
someone snuck in submodule ref changes ?

willthames commented:
@bcoca, oops, I updated my commit and it pulled those in. I've removed them now. 'snuck in' is probably an ungenerous term.

bcoca commented:
sorry, i get cranky after a full day of triage

willthames commented:
ready_for_review

mattclay commented on lib/ansible/parsing/mod_args.py:
```diff
@@ -178,15 +178,15 @@ def _normalize_parameters(self, thing, action=None, additional_args=dict()):
-        deals with fuzziness in old-style (action/local_action) module invocations
+        deals with fuzziness in new style module invocations
+        accepting key=value pairs and dictionaries, and always returning dictionaries
         returns tuple of (module_name, dictionary_args)
```
The return comment doesn't match the return value. Not a new problem, but should be fixed.

mattclay commented on lib/ansible/parsing/mod_args.py:
```diff
@@ -178,15 +178,15 @@ def _normalize_parameters(self, thing, action=None, additional_args=dict()):
+            'echo hi', 'shell'
+            {'region': 'xyz'}, 'ec2'
         standardized outputs like:
             ( 'command', { _raw_params: 'echo hi', _uses_shell: True }
```
This output comment is incorrect. Also not a new problem, but fix it while you're in the area.

mattclay reviewed (changes requested):
I spotted two other corrections that should be made to the comments.

willthames committed:
Further improve mod_args.py comments

Ensure output formats are correctly documented,
remove some of the 'opinion' about which formats are
valid, and try and clarify the situations under which
certain code paths are hit.

Stop talking about the YAML command-type form as 'extra
gross' when it's the documented example form for command
etc.!

willthames commented:
!needs_revision

willthames commented:
ready_for_review

willthames commented:
@mattclay I fixed your review points and also went a bit further

mattclay commented:
@willthames Merged! Thanks for cleaning this up.
